File: SampleSolution.Core/Models/ViewModels/PaginationViewModels/PageMeta.cs
namespace SampleSolution.Core.Models.ViewModels.Pagination
{
    public class PageMeta
    {
        public long PageNumber { get; set; } = 0;
        public long PageSize { get; set; } = 0;
        public long ElementsInPage { get; set; } = 0;
        public long TotalPages { get; set; } = 0;
        public long TotalElements { get; set; } = 0;
    }
}

File: Clients/SampleSolution.Web/wwwrootsrc/core/scripts/models/ViewModels/PageMeta.ts
import { jsonProperty, Serializable } from "ts-serializable";

export class PageMeta extends Serializable {

    @jsonProperty(Number)
    public pageNumber: number = 0;

    @jsonProperty(Number)
    public pageSize: number = 0;

    @jsonProperty(Number)
    public elementsInPage: number = 0;

    @jsonProperty(Number)
    public totalPages: number = 0;

    @jsonProperty(Number)
    public totalElements: number = 0;

}
